src/main/java/appeng/util/item/ItemList.java

```diff
@@ -25,17 +25,73 @@
 import java.util.concurrent.ConcurrentSkipListMap;
 import net.minecraftforge.oredict.OreDictionary;
 
-public final class ItemList implements IItemList<IAEItemStack> {
-
+public class ItemList implements IItemList<IAEItemStack> {
     private final NavigableMap<IAEItemStack, IAEItemStack> records =
             new ConcurrentSkipListMap<IAEItemStack, IAEItemStack>();
+    private final Cache cache;
+
+    public ItemList(boolean useCache) {
+        this.cache = new Cache(useCache);
+    }
+
+    public ItemList() {
+        this.cache = new Cache(false);
+    }
+
+    private static class Cache {
+        private final HashMap<Integer, Collection<IAEItemStack>> cache;
+        private final HashMap<Integer, Boolean> updated;
+        private final boolean useCache;
+        private final Collection<IAEItemStack> emptyCollection = new LinkedList<IAEItemStack>();
```

Comment:
Don't use linked lists, they use much more memory and are slower than ArrayLists